### PR TITLE
[2.4] Improve cell pipe timeout handling

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # 4572538.

### Description

Currently pipe timeout replies on heartbeat messages. Even though heartbeats are sent in an independent thread, when the pipe is busy with large task data (e.g. 8G), the heartbeat messages will be queued after some chunks of task data. It will take time for the heartbeat messages to arrive, which could be too late (> configured timeout value).

This PR tries to improve it by intercepting all messages from the peer and remember the last time that any message is received from the peer. This time is used in the timeout logic: if it is > the last heartbeat time, it is treated as the last HB time.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/fuel/utils/pipe/cell_pipe.py [hunk 1]
```diff
@@ -249,6 +292,9 @@ def send(self, msg: Message, timeout=None) -> bool:
                         self.logger.error(err)
                     return False
             else:
+                self.logger.error(
```
Comment (SYangster) [inline nvflare/fuel/utils/pipe/cell_pipe.py:295]: add f-string for format

File: nvflare/fuel/utils/pipe/cell_pipe.py [hunk 2]
```diff
@@ -46,8 +49,10 @@ def _cell_fqcn(mode, site_name, token):
     return f"{site_name}_{token}_{mode}"
 
 
-def _to_cell_message(msg: Message) -> CellMessage:
-    headers = {_HEADER_MSG_TYPE: msg.msg_type, _HEADER_MSG_ID: msg.msg_id}
+def _to_cell_message(msg: Message, extra=None) -> CellMessage:
+    headers = {_HEADER_MSG_TYPE: msg.msg_type, _HEADER_MSG_ID: msg.msg_id, _HEADER_START_TIME: time.time()}
```
Comment (SYangster) [inline nvflare/fuel/utils/pipe/cell_pipe.py:53]: `_HEADER_START_TIME` doesn't seem to be used anywhere?

Comment (YuanTingHsieh) [reply nvflare/fuel/utils/pipe/cell_pipe.py:53]: If `_HEADER_START_TIME` is for debugging purpose, should we pass it in as `extra_headers = {_HEADER_HB_SEQ: self.hb_seq}`?

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM, 

This PR change control signals (ABORT/END/HB) handling, instead of waiting for reply (which might takes a long time):
- This PR use fire_and_forget for HB
- This PR just check if sending is done for ABORT/END

This PR also enhances the timeout/keep_alive mechanism, if one side has received anything (reply/request) from the other side, the active time will be updated.

File: nvflare/fuel/utils/pipe/cell_pipe.py (hunk 2 above)
Comment (yanchengnv) [reply nvflare/fuel/utils/pipe/cell_pipe.py:53]: It is

**Pull request merged**